### ISSUE TITLE
UnitarySystem with SingleZoneVAV control crashes

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#7085]
Issue overview
--------------
While adding SingleZoneVAV control to `HVACTemplate:System:UnitarySystem` ran into a crash with DX coil.
Also crashes with chilled water coil - in fact the first test file may be crashing on the chilled water one, not the DX.

### Details
Some additional details for this issue (if relevant):
 - Version of EnergyPlus 9.0.1 must be a debug build to catch the crash
 - Reference #7062 

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect files
[7085-HVACTemplate-5ZoneUnitarySystem-SinglZoneVAV-exp-DXandChW.idf.txt](https://github.com/NREL/EnergyPlus/files/2640735/7085-HVACTemplate-5ZoneUnitarySystem-SinglZoneVAV-exp-DXandChW.idf.txt)
[7085-HVACTemplate-5ZoneUnitarySystem-SinglZoneVAV-exp-AllChW.idf.txt](https://github.com/NREL/EnergyPlus/files/2640736/7085-HVACTemplate-5ZoneUnitarySystem-SinglZoneVAV-exp-AllChW.idf.txt)

 added here: 

 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by mjwitte** [comment]
Also, can't find any example files that use single zone VAV either for unitary or fan coil?

**Comment by rraustad** [comment]
Divide by 0 with water coil water flow rate. E+ does allow 0 flow for water coils so this needs to be protected. Don't even need to do SZVAV calc's if max flow = 0.

    SZVAVModel.HeatCoilWaterFlowRatio = maxCoilFluidFlow / SZVAVModel.MaxHeatCoilFluidFlow;

**Comment by rraustad** [comment]
Protection allows the file to run. Now I should figure out why Sys 1 and 2 have 0 cooling capacity and water flow when those zones have a design load.

    Component Sizing Information, Coil:Cooling:Water, SYS 1 FURNACE DX COOL SNGLSPD COOLING COIL, Design Size Design Coil Load [W], 0.00000
    Component Sizing Information, Coil:Cooling:Water, SYS 1 FURNACE DX COOL SNGLSPD COOLING COIL, Design Size Design Water Flow Rate [m3/s], 0.00000

    Component Sizing Information, Coil:Cooling:Water:DetailedGeometry, SYS 2 FURNACE DX COOL MULTISPD COOLING COIL, Design Size Design Coil Load [W], 0.00000
    Component Sizing Information, Coil:Cooling:Water:DetailedGeometry, SYS 2 FURNACE DX COOL MULTISPD COOLING COIL, Design Size Maximum Water Flow Rate [m3/s], 0.00000

**Comment by rraustad** [comment]
Not sure why this hasn't shown up before using autosize for a water coil in a UnitarySystem? Need to check other example files for use of water coils.

    Component Sizing Information, Coil:Cooling:Water, SYS 1 FURNACE DX COOL SNGLSPD COOLING COIL, Design Size Design Coil Load [W], 7576.42152
    Component Sizing Information, Coil:Cooling:Water, SYS 1 FURNACE DX COOL SNGLSPD COOLING COIL, Design Size Design Water Flow Rate [m3/s], 2.70612E-004

**Comment by rraustad** [comment]
The 0 sized coil problem was that the water coils don't have a capacity input field. I always wondered what to do about that during sizing. To correct I simply added Capacity = autosize in GetInput for coil types without a capacity input field. Then I wondered if I did that what would happen if a user hard-sized the water flow rate. So as a test I took the defect file and halved the water flow to see what happened (from autosize = 3.88275E-004 to manual size of 1.98275E-4). The coil capacity is roughly the same and the UA and surface area have increased significantly (about 3 fold, last 2 fields in Water Cooling Coil Capacity Information). This seems reasonable since the coil still needs to know the load to size the coil and with half the flow these other parameters need to increase.

![image](https://user-images.githubusercontent.com/5778156/55558964-2c15a580-56bb-11e9-86f3-87cf6b1df7da.png)
